packages/editor/src/services/property.util.ts

```diff
@@ -0,0 +1,307 @@
+  const schemaRefs = JsonRefs.findRefs(schema, {resolveCirculars: true});
+  const allRefs = JsonRefs.findRefs(parentSchema, {resolveCirculars: true});
+  let extractedReferences;
+  findReferences(parentSchema, allRefs, schemaRefs, extractedReferences = {});
+  const refList = _.values(extractedReferences) as string[];
+  if (!_.isEmpty(refList)) {
+    _.each(refList, ref => {
+      const propertyKey = ref.substring((ref.indexOf('/') + 1), (ref.lastIndexOf('/')));
+      const property = ref.substring((ref.lastIndexOf('/') + 1));
+      if (_.has(parentSchema, propertyKey) && _.has(parentSchema, `${propertyKey}.${property}`)) {
+        if (schema[propertyKey]) {
+          schema[propertyKey] = {
+            ...schema[propertyKey], ...{
+              [property]: parentSchema[propertyKey][property]
+            }
+          };
+        } else {
+          schema = {
+            ...schema, ...{
+              [propertyKey]: { [property]: parentSchema[propertyKey][property] }
+            }
+          };
+        }
+      }
+    });
+  }
+
+  return schema;
+};
+
+/**
+ * Create a self contained schema.
+ * @param parentSchema The schema to use for resolving
+ * @param refPath The path to resolve
+ * @return a JsonSchema that is self-contained
+ */
+export const makeSelfContainedSchema = (parentSchema: JsonSchema, refPath: string): JsonSchema => {
+  let schema = resolveSchema(parentSchema, refPath);
+  schema = deepCopy(schema);
+  if (_.isEmpty(schema.id)) {
+    schema.id = '#' + refPath;
+  }
+  schema = { ...schema, ...makeSchemaSelfContained(parentSchema, schema) };
+
+  return schema;
+};
+
+/**
+ * Calculates label for an item which is resolved in anyOf
+ * @param {string} id The schema id
+ * @returns {string}
+ */
+const calculateLabel = (id: string): string => (
+  _.startsWith(id, '#') ? id.substring(id.lastIndexOf('#') + 1) : id
+);
+
+/**
+ * Constructing container properties of a given schema. A container means
+ * a given property can hold a set of its sub-properties. This allows to create
+ * a hierarchy of elements. If the given schema isn't a array type of object, it is
+ * not considered as a container. Since only array elements can be added into the
+ * parent property, all other properties that don't contain array type of object
+ * are discarded.
+ *
+ * @param {string} property The schema key from which this property was created.
+ * @param {string} label The name of the schema the property describes.
+ * @param {JsonSchema} schema The schema is the JsonSchema this property describes.
+ * @param {JsonSchema} rootSchema The parent schema
+ * @param {boolean} isInContainer To indicate whether the schema is in a container or not
+ *                                Properties that are described in array are considered as
+ *                                they are in a container.
+ * @param {boolean} hasOnlyOwnChildren To accept only corresponding children of a property
+ * @returns {Property[]}
```

Comment:
Describe return type and mention that the returned properties all have a self-contained schema.